pkg/agent/config.go

```diff
@@ -14,9 +14,25 @@ const (
 
 type AgentSpec struct {
 	Metadata AgentMetadata `json:"metadata"`
+	Builtin  *BuiltinRef   `json:"builtin,omitempty"`
 	Commands AgentCommands `json:"commands"`
 }
 
+// BuiltinRef references a built-in agent type with optional model
+type BuiltinRef struct {
+	// Type is the built-in agent type (e.g., "openai-agent", "claude-code")
+	Type string `json:"type"`
+
+	// Model is the AI model to use (required for some types like openai-agent)
+	Model string `json:"model,omitempty"`
+
+	// BaseURL overrides the default API base URL
+	BaseURL string `json:"baseUrl,omitempty"`
+
+	// APIKey overrides the default API key (from environment)
+	APIKey string `json:"apiKey,omitempty"`
```

Comment:
Not directly related - but I guess at some point we should use a more consistent naming
`MODEL_BASE_URL` vs `BASE_URL`

or `MODEL_KEY` vs `API_KEY` 
CC @Cali0707